axelor-production/src/main/java/com/axelor/apps/production/web/UnitCostCalculationController.java

```diff
@@ -131,4 +134,69 @@ public void importUnitCostCalc(ActionRequest request, ActionResponse response)
       TraceBackService.trace(response, e);
     }
   }
+
+  @SuppressWarnings("unchecked")
+  public void fillProductSetDomain(ActionRequest request, ActionResponse response) {
+    try {
+      UnitCostCalculation unitCostCalculation =
+          request.getContext().asType(UnitCostCalculation.class);
+
+      Company company = null;
+
+      if (Beans.get(UnitCostCalculationService.class).hasDefaultBOMSelected()) {
+        LinkedHashMap<String, Object> companyMap =
+            (LinkedHashMap<String, Object>) request.getContext().get("company");
+        if (companyMap != null) {
+          company =
+              Beans.get(CompanyRepository.class).find(((Integer) companyMap.get("id")).longValue());
+        }
+      }
+
+      String domain =
+          Beans.get(UnitCostCalculationService.class)
+              .createProductSetDomain(unitCostCalculation, company);
+      response.setAttr("productSet", "domain", domain);
+
+    } catch (Exception e) {
+      TraceBackService.trace(response, e, ResponseMessageType.ERROR);
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  public void fillCompanySet(ActionRequest request, ActionResponse response) {
+    if (Beans.get(UnitCostCalculationService.class).hasDefaultBOMSelected()) {
+      UnitCostCalculation unitCostCalculation =
+          request.getContext().asType(UnitCostCalculation.class);
+
+      unitCostCalculation =
+          Beans.get(UnitCostCalculationRepository.class).find(unitCostCalculation.getId());
+
+      LinkedHashMap<String, Object> companyMap =
+          (LinkedHashMap<String, Object>) request.getContext().get("company");
+
+      if (companyMap == null) {
+        return;
+      }
+
+      Company company =
+          Beans.get(CompanyRepository.class).find(((Integer) companyMap.get("id")).longValue());
+
+      Beans.get(UnitCostCalculationService.class).fillCompanySet(unitCostCalculation, company);
+    }
+  }
+
+  public void fillCompany(ActionRequest request, ActionResponse response) {
+    if (Beans.get(UnitCostCalculationService.class).hasDefaultBOMSelected()) {
+      UnitCostCalculation unitCostCalculation =
+          request.getContext().asType(UnitCostCalculation.class);
+
+      unitCostCalculation =
+          Beans.get(UnitCostCalculationRepository.class).find(unitCostCalculation.getId());
+
+      Company company =
+          Beans.get(UnitCostCalculationService.class).getSingleCompany(unitCostCalculation);
+
+      response.setValue("$company", company);
+    }
```

Comment:
Please put everything in a try {} catch {} to trace any exception. Also please call Beans.get(UnitCostCalculationService.class) only once